drivers/bridge/setup_ip_tables.go

```diff
@@ -46,17 +50,49 @@ func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainI
 			}
 		}
 	}()
+	if err := ensureReturnRule(DockerChain); err != nil {
+		return err
+	}
 
-	isolationChain, err := iptables.NewChain(IsolationChain, iptables.Filter, false)
+	d.exposedChain, err = iptables.NewChain(ExposedChain, iptables.Filter, false)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to create FILTER isolation chain: %v", err)
+		return fmt.Errorf("failed to create FILTER chain: %v", err)
+	}
+	defer func() {
+		if err != nil {
+			if err := iptables.RemoveExistingChain(ExposedChain, iptables.Filter); err != nil {
+				logrus.Warnf("failed on removing iptables FILTER chain on cleanup: %v", err)
+			}
+		}
+	}()
+	if err := ensureReturnRule(ExposedChain); err != nil {
+		return err
 	}
 
-	if err := addReturnRule(IsolationChain); err != nil {
-		return nil, nil, nil, err
+	d.isolationChain, err = iptables.NewChain(IsolationChain, iptables.Filter, false)
+	if err != nil {
+		return fmt.Errorf("failed to create FILTER isolation chain: %v", err)
+	}
+	defer func() {
+		if err != nil {
+			if err := iptables.RemoveExistingChain(IsolationChain, iptables.Filter); err != nil {
+				logrus.Warnf("failed on removing iptables FILTER chain on cleanup: %v", err)
```

Comment:
Same here, and actually for the other `defer func()` which are in master (not part of your PR).